test/jdk/java/util/HashMap/WhiteBoxResizeTest.java

```diff
@@ -21,113 +22,588 @@
+                pcc("phmcpy", size, cap, () -> new HashMap<>(makeMap(size)), map -> {
+                }),
+                pcc("phm0pn", size, cap, () -> new HashMap<>(), map -> {
+                    putN(map, size);
+                }),
+                pcc("phm1pn", size, cap, () -> new HashMap<>(cap), map -> {
+                    putN(map, size);
+                }),
+                pcc("phm2pn", size, cap, () -> new HashMap<>(cap, 0.75f), map -> {
+                    putN(map, size);
+                }),
+                pcc("phm0pa", size, cap, () -> new HashMap<>(), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("phm1pa", size, cap, () -> new HashMap<>(cap), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("phm2pa", size, cap, () -> new HashMap<>(cap, 0.75f), map -> {
+                    map.putAll(makeMap(size));
+                }),
+
+                pcc("plmcpy", size, cap, () -> new LinkedHashMap<>(makeMap(size)), map -> {
+                }),
+                pcc("plm0pn", size, cap, () -> new LinkedHashMap<>(), map -> {
+                    putN(map, size);
+                }),
+                pcc("plm1pn", size, cap, () -> new LinkedHashMap<>(cap), map -> {
+                    putN(map, size);
+                }),
+                pcc("plm2pn", size, cap, () -> new LinkedHashMap<>(cap, 0.75f), map -> {
+                    putN(map, size);
+                }),
+                pcc("plm0pa", size, cap, () -> new LinkedHashMap<>(), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("plm1pa", size, cap, () -> new LinkedHashMap<>(cap), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("plm2pa", size, cap, () -> new LinkedHashMap<>(cap, 0.75f), map -> {
+                    map.putAll(makeMap(size));
+                }),
+
+                pcc("pwmcpy", size, cap, () -> new WeakHashMap<>(makeMap(size)), map -> {
+                }),
+                pcc("pwm0pn", size, cap, () -> new WeakHashMap<>(), map -> {
+                    putN(map, size);
+                }),
+                pcc("pwm1pn", size, cap, () -> new WeakHashMap<>(cap), map -> {
+                    putN(map, size);
+                }),
+                pcc("pwm2pn", size, cap, () -> new WeakHashMap<>(cap, 0.75f), map -> {
+                    putN(map, size);
+                }),
+                pcc("pwm0pa", size, cap, () -> new WeakHashMap<>(), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("pwm1pa", size, cap, () -> new WeakHashMap<>(cap), map -> {
+                    map.putAll(makeMap(size));
+                }),
+                pcc("pwm2pa", size, cap, () -> new WeakHashMap<>(cap, 0.75f), map -> {
+                    map.putAll(makeMap(size));
+                })
+        );
```

Comment:
Wait, did this get reindented? Adding line breaks totally destroys the tabular nature of test data. Please restore. Yes, the lines end up being longer than the usual limit, but the benefit of having things in a nice table outweighs to cost of having the lines be somewhat over-limit.

Reply:
@stuart-marks done.